cdvae/pl_modules/gnn.py

```diff
@@ -340,51 +346,42 @@ def __init__(
             num_output_layers=num_output_layers,
         )
 
-    def forward(self, data):
+    def _forward(self, data):
+        pos = data.pos
         batch = data.batch
-
-        if self.otf_graph:
-            edge_index, cell_offsets, neighbors = radius_graph_pbc_wrapper(
-                data, self.cutoff, self.max_num_neighbors, data.num_atoms.device
-            )
-            data.edge_index = edge_index
-            data.to_jimages = cell_offsets
-            data.num_bonds = neighbors
-
-        pos = frac_to_cart_coords(
-            data.frac_coords,
-            data.lengths,
-            data.angles,
-            data.num_atoms)
-
-        out = get_pbc_distances(
-            data.frac_coords,
-            data.edge_index,
-            data.lengths,
-            data.angles,
-            data.to_jimages,
-            data.num_atoms,
-            data.num_bonds,
-            return_offsets=True
-        )
-
-        edge_index = out["edge_index"]
-        dist = out["distances"]
-        offsets = out["offsets"]
-
+        (
+            edge_index,
+            dist,
+            _,
+            cell_offsets,
+            offsets,
+            neighbors,
+        ) = self.generate_graph(data)
+
+        data.edge_index = edge_index
+        data.cell_offsets = cell_offsets
+        data.neighbors = neighbors
         j, i = edge_index
 
         _, _, idx_i, idx_j, idx_k, idx_kj, idx_ji = self.triplets(
-            edge_index, num_nodes=data.atom_types.size(0)
+            edge_index,
+            data.cell_offsets,
```

Comment:
what are these `cell_offsets`? is `cell` analogous to pbc?

Reply:
all good; set to 0 in `generate_graph` logic (because `use_pbc` and `otf_graph` are False)